src/board.cpp

```diff
@@ -183,6 +286,131 @@ void Board::removePiece(Position pos)
     board[pos.row][pos.col] = 32;
 }
 
+
+
+// add a piece to the board, should only be used in the constructor for blank boards
+ void Board::addPiece(const pieceLocationInput& inputPiece)
+ {
+    if (inputPiece.PieceColor == Color::EMPTY)
+        throw invalid_argument("bad color for added piece");
+
+    if (board[inputPiece.piecePos.row][inputPiece.piecePos.col] != 32)
+        throw invalid_argument("trying to add piece to a non empty square");
+
+    // array that will contain a list of all possible ids for a specific pieceKind/Color combination. I am giving it a size of nine so that if we get to the end it spits out a -1
+    uint8_t idsArray[9] = {32,32,32,32,32,32,32,32,32};
+    
+    
+    int size = reversePieceLookup(inputPiece.pieceKind, inputPiece.PieceColor,idsArray);
+
+    int currentIDIndex = 0; // index pointing to the id we think is not taken
+
+
+    // going through the whole board array
+    for (int row = 0; row < 8; ++row)
+    {
+        for (int col = 0; col < 8; ++col)
+    {
+
+        // checking if our possible id's array contains the id of this square
+        for (int loopCount = 0; loopCount < size; ++loopCount)
+        {
+            if (board[row][col] == idsArray[loopCount])
+            {
+                idsArray[loopCount] = 32;
+                break;
+            }
+        }
+    }
+    }
+
+
+    for (currentIDIndex; currentIDIndex < size; ++currentIDIndex)
```

Comment:
The first statement in a for loop is the initializer for the loop variable. If you don't need to initialize it, leave that first statement blank. 
```suggestion
    for (; currentIDIndex < size; ++currentIDIndex)
```